contracts/usul/BaseQuorumPercent.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+/// @title BaseQuorumPercent - A Usul strategy extension that enables percent based quorums
+abstract contract BaseQuorumPercent is OwnableUpgradeable {
+    uint256 public quorumNumerator;
+    uint256 public constant quorumDenominator = 1_000_000;
+
+    event QuorumNumeratorUpdated(uint256 newQuorumNumerator);
+
+    function quorum(uint256 blockNumber) public view virtual returns (uint256);
+
+    function updateQuorumNumerator(
+        uint256 newQuorumNumerator
+    ) public virtual onlyOwner {
+        _updateQuorumNumerator(newQuorumNumerator);
+    }
```

Comment:
another instance I'm not sure it's necessary to have `public` / `internal` versions of, it seems like we could just have a single `external` function

Reply:
So yeah this is also just to DRY the code up. `LinearTokenVoting` calls the internal `_updateQuorumNumerator` from within its `setUp` function.